### PR TITLE
Budget_app

### PR DESCRIPTION
**Pull request by Kel-nana**
### - Interactions
- Splash screen

1. A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

Sign up and log in pages

2. The user should be able to register in the app with full name, email and password (all mandatory).

3. The user can log into the app using email and password.
If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
- Home page (categories page)

1. When the user logs in, they are presented with the categories page.
For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
- Transactions page

1. For a given category, the list of transactions is presented, ordered by the most recent.
At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
2. There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
3. When the user clicks on the "Back" button (<), the user navigates to the home page.
 "Add a new category" page

5. The user fills out a form to create a new category, indicating their name and icon (both mandatory).
6. he user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
7. When the user clicks on the "Back" button (<), the user navigates to the home page.
- "Add a new transaction" page

1. The user fills out a form to create a new transaction with:
2. name (mandatory)
3. amount (mandatory)
4. categories (mandatory at least one)
5. The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
6. When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
- Testing requirements
- Create unit and integration tests for all the most important components of your RoR application.
### - Technical requirements
- You should use Postgres as your database.
- You should use devise for authentication.
- You should validate all user input to make sure that anyone with bad intentions cannot compromise your app.
- The project should be deployed and accessible online.
- Your database schema should reflect the following structure:

### CODE REVIEW

File: app/views/categories/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+  <section class="text-center mt-4">
+    <%= link_to new_category_path do %>
+      <div class="w-full green-btn flex">
+        <button class="p-2 py-4 m-4 text-lg w-full border-2 green-btn rounded hover:bg-green-600 focus:outline-none focus:bg-green-600">
+          <strong>Add a New Category</strong>
+        </button>
+      </div>
+    <% end %>
+  </section>
```
Comment (elisha2kyakpo1) [inline app/views/categories/index.html.erb]: - [ ] Your design is good, but there are some minor changes that should be done on the categories and the transactions pages respectively. The add categories button should touch the two ends (left and right), the same should be done on the categories.


button


spaces

File: app/views/purchases/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  <ul class="display_nav flex justify-between items-center ">
+    <li class="w-10"><%= link_to image_tag('backarrow.png', alt: 'Back'), categories_path, class: 'nav-link' %></li>
+    <h2><strong>New purchase</strong></h2>
+    <li><%= link_to 'Sign Out', destroy_user_session_path, data: { turbo_method: :delete }, class: "nav-link text-green" %></li>
+  </ul>
+<div class="mt-10 ml-12 flex flex-col gap-5 ms-4 justify-center content-center  flex-grow">
+<%= render "shared/form_purchase", attributes: @attributes %>
+ | <%= link_to "home purchase", purchases_path,class:"text-green" %>
+</div>
+</div>
```
Comment (elisha2kyakpo1) [inline app/views/purchases/new.html.erb]: - [ ] Kindly try as much as you can to make this page look like the given design. The button should be as big as the add categories button and should be positioned at the bottom. Pay attention to spaces for the purchases and the button.


button


btn

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @Kel-nana, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description are good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.
✔️ Beautiful UI👏
👏 The app works as expected.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mahbubul14 — APPROVED** [review]
# Project Approved :trophy: :tada: 

Hi @Kel-nana, 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linter error.
:heavy_check_mark: Descriptive readme file.
:heavy_check_mark: Well implemented reviewer's suggestions.

### Optional suggestions :arrow_down_small: 

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

- [x] The commit message should be written in the imperative tense and the first letter should also be capitalized. `Ex: "Fix bug" instead of "Fixing bug".` [For more details...](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**